cpp/cmake/modules/embed_fatbins.cmake

```diff
@@ -0,0 +1,57 @@
+# =============================================================================
+# Copyright (c) 2025, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+# =============================================================================
+
+function(embed_fatbins library_name kernel_target)
+  find_package(CUDAToolkit REQUIRED)
+  find_program(
+    bin_to_c
+    NAMES bin2c
+    PATHS ${CUDAToolkit_BIN_DIR}
+  )
+
+  set(output_dir ${CMAKE_CURRENT_BINARY_DIR}/${library_name})
+
+  # Create a response file to avoid "argument list too long" errors
+  set(objects_response_file ${CMAKE_CURRENT_BINARY_DIR}/${library_name}_objects.rsp)
```

Comment:
Place the `rsp` file into a subdirectory with a name that makes it clear what has generated it:
```
 ${CMAKE_CURRENT_BINARY_DIR}/embed_fatbins/${library_name}_objects.rsp
 ```